### PR TITLE
moved more types into common types package

### PR DESCRIPTION
**Pull request by david-driscoll**
This is a proposal for #57.

Feel free to let me know what may / may not need to change.  :tada:

### CODE REVIEW

**Comment by hediet** [comment]
Can we please have a single source of every type (including request and notification type specifications) that is mentioned in the language server protocol specification and separate them from types not mentioned there?
Right now, it is just incredibly hard to generate code for other languages, as the specification is very inconsistent and thus hard to parse (its typescript even has syntactical errors) and the types here are mixed with implementation types that have nothing to do with the protocol (like `TextEditChange` or `TextDocument`).
Your pull request makes it even harder, as it adds another indirection with all member in the `Methods` class - getting the method out of `{ get method() { return 'textDocument/hover'; } };` is already hard enough when using the typescript parser API.

**Comment by dbaeumer** [comment]
The idea of the types package is to have the types which are independent of the protocol in a separate package. So we should not move protocol specific types in there. @david-driscoll why do you need these types in a separate package. Do you want to implement your own client in TS ?

@hediet the idea for this is to have a JSON schema in the protocol repository which then can be used to generate 'interfaces' for other languages.

@aeschli any additional comment.

**Comment by david-driscoll** [comment]
Yeah, I'm implementing the client for Atom.  Almost all of the types are reusable at the client level.  I hate to have to copy and paste things, but that's what I have at the moment.

**Comment by aeschli** [comment]
+1 for a JSON schema description of all type protocol messages & types.

Maybe we should think of a language-service-module package, containing protocol types/messages plus a Connection implementation that can be used for server and client implementations, without any VSCode specific bindings. That would help David as well as other language client implementations that want to implement a client in NodeJS.

**Comment by dbaeumer** [comment]
All these types are now available in the protocol npm module which is independent of server or client.
